### PR TITLE
Update cypress-react-unit-test, run desktop-gui component tests

### PR DESCRIPTION
**Pull request by bahmutov**
- internal tests
Added component test that shows how simple it is to work with Desktop GUI components. Stub common imports, set up surrounding DOM structure, interact with the component and take visual screenshots

Screen Shot 2020-06-23 at 4 55 22 PM

## To resolve

- [x] link separate test jobs into a single Percy run (right now desktop-gui creates 2 Percy builds)
  * https://docs.percy.io/docs/parallel-test-suites

Because we have 2 jobs that create Percy snapshots added to both the following

```
PERCY_PARALLEL_NONCE=$CIRCLE_WORKFLOW_ID \
PERCY_PARALLEL_TOTAL=2 \
```

and now all screenshots are in the same build

## Dropped

I will omit Percy from component testing for now, since this problem is still open. Will investigate it in https://github.com/bahmutov/cypress-react-unit-test/issues/313 instead and come back to enable Percy snapshots for component tests here

- [ ] component test in Percy is missing styles

Screen Shot 2020-06-24 at 3 05 20 PM

Playing with caching setting in https://github.com/cypress-io/cypress/blob/develop/.percy.yml did not help. This is the structure of the test DOM

Screen Shot 2020-06-24 at 3 26 48 PM

The CSS is there

Screen Shot 2020-06-24 at 3 27 14 PM

### Why

I think this happens because these components get they style from `/__root/...` URL served by Cypress itself, see `stylesheets` parameter

```js
mount(
  <div className="settings">
    <div className="settings-wrapper">
      <Collapse>
        <Panel header='File Opener Preference' key='file-preference' className='form-horizontal settings-file-preference'>
          <FilePreference />
        </Panel>
      </Collapse>
    </div>
  </div>,
  { 
    alias: 'FilePreference', 
    stylesheets: '/__root/dist/app.css' 
  },
)
```

Percy runs like this:

```
percy-exec -- cypress run ...
```

Seems Percy waits for Cypress to exit before trying to get the CSS ... and then the URL does not respond any more.

### CODE REVIEW

**Review by panzarino — COMMENTED** [review]
I feel like it would make sense to write tests like this for all components in this package before merging in just a single test.

One of my PRs will modify a bunch of things in `ui-components`, so it might be best to wait for that to be merged in: https://github.com/cypress-io/cypress/pull/7747

**Comment by bahmutov** [comment]
@panzarino that is the wrong way to go about it. Upgrading cypress-react-unit-test by itself is a good thing, and every PR should be focused, rather than a giant "write tests for all components" kind of thing. 

Also, I have nothing but curse words for TypeScript and lock file that springs unrelated flood of errors for no reason

**Comment by panzarino** [comment]
@bahmutov True. I misread the files during initial review and thought this was adding to `ui-components` which is why I suggested writing tests for all the components (because there are only a couple and no unit tests yet).

I think this would be good to merge in assuming you figure out the TS stuff.

**Review by panzarino — APPROVED** [review]

**Pull request merged**
